### PR TITLE
Add {Batch,batcher}.GetProto.

### PR DESCRIPTION
**Pull request by petermattis**
Added a simple test for a set of common methods between DB, Tx, Batch
and batcher.

### CODE REVIEW

File: client/db_test.go [hunk 1]
```diff
@@ -288,3 +289,31 @@ func TestDebugName(t *testing.T) {
 		return nil
 	})
 }
+
+func TestCommonMethods(t *testing.T) {
+	methods := []string{
```
Comment (bdarnell) [inline client/db_test.go]: A test based on whitelisting the methods that are expected to exist everywhere won't help much - it won't catch new methods like GetProto unless you know to add them to the list. I think we need a blacklist instead, of methods that are allowed to differ between the different types.

**Comment by bdarnell** [comment]
LGTM

File: client/db_test.go (hunk 1 above)
Comment (petermattis) [inline client/db_test.go]: Ok, switched this around to using a blacklist instead. Not a very efficient implementation, but that doesn't matter.

File: client/db_test.go [hunk 2]
```diff
@@ -288,3 +290,46 @@ func TestDebugName(t *testing.T) {
...
 }
+
+func TestCommonMethods(t *testing.T) {
+	types := []reflect.Type{
+		reflect.TypeOf(&client.DB{}),
+		reflect.TypeOf(&client.Tx{}),
+		reflect.TypeOf(&client.Batch{}),
+		reflect.TypeOf(client.DB{}.B),
+	}
+	blacklist := map[string]bool{
```
Comment (tamird) [inline client/db_test.go]: nit: i thought the canonical set was a `map[myType]struct{}`

Comment (petermattis) [inline client/db_test.go]: It is, but I don't think the space savings outweighs the ease of using `blacklist[name]`.

File: client/db_test.go [hunk 3]
```diff
@@ -288,3 +290,46 @@ func TestDebugName(t *testing.T) {
...
+		"AdminMerge":          true,
+		"AdminSplit":          true,
+		"Commit":              true,
+		"DebugName":           true,
+		"InternalAddCall":     true,
+		"InternalKV":          true,
+		"InternalSetPriority": true,
+		"Run":                  true,
```
Comment (petermattis) [inline client/db_test.go]: Not sure why `gofmt` is producing this funny indentation. But it is doing so repeatedly.

File: client/db_test.go (hunk 2 above)
Comment (tamird) [inline client/db_test.go]: ah, right you are. one more nit: can you add something that makes sure things on the blacklist exist on at least one of `types`?

File: client/db_test.go [hunk 4]
```diff
@@ -288,3 +290,46 @@ func TestDebugName(t *testing.T) {
...
+		"SetDebugName":         true,
+		"SetSnapshotIsolation": true,
+		"Tx": true,
+	}
+	for _, typ := range types {
+		for j := 0; j < typ.NumMethod(); j++ {
+			m := typ.Method(j)
+			var ch rune
```
Comment (bdarnell) [inline client/db_test.go]: `ch, _ := utf8.DecodeRune(m.Name)`

Or, according to http://golang.org/pkg/reflect/#Method, `m.PkgPath == ""` iff `m` is exported.

File: client/db_test.go (hunk 2 above)
Comment (petermattis) [inline client/db_test.go]: I can actually make the blacklist type specific.

File: client/db_test.go [hunk 5]
```diff
@@ -288,3 +290,46 @@ func TestDebugName(t *testing.T) {
...
+			for _, ch = range m.Name {
```
Comment (tamird) [inline client/db_test.go]: so this just gets the first character?

Comment (tamird) [inline client/db_test.go]: wouldn't `[]rune(m.Name)[0]` work?

File: client/db_test.go (hunk 2 above)
Comment (bdarnell) [inline client/db_test.go]: It's not about space savings IMO, it's the fact that future developers don't have to think about whether there is a difference between a key that is not present and a key that is set to false. We use `map[T]struct{}` pretty consistently.

File: client/db_test.go (hunk 5 above)
Comment (petermattis) [inline client/db_test.go]: Does casting a `string` to a `[]rune` work?

Comment (bdarnell) [inline client/db_test.go]: Yes. It decodes the whole string (and allocates for the result) so it's a bit less efficient than utf8.DecodeRune, but it's probably the simplest option and performance doesn't matter here.

Comment (tamird) [inline client/db_test.go]: I think the pkgpath thing is probably the best

File: client/db_test.go (hunk 2 above)
Comment (petermattis) [inline client/db_test.go]: Fine, switched to `map[string]struct{}`.

File: client/db_test.go (hunk 5 above)
Comment (petermattis) [inline client/db_test.go]: Huh, I learned something new about strings and rune slices. Changed to using `Method.PkgPath`.

File: client/db_test.go [hunk 6]
```diff
@@ -288,3 +289,46 @@ func TestDebugName(t *testing.T) {
...
 }
+
+func TestCommonMethods(t *testing.T) {
+	types := []reflect.Type{
+		reflect.TypeOf(&client.DB{}),
+		reflect.TypeOf(&client.Tx{}),
+		reflect.TypeOf(&client.Batch{}),
+		reflect.TypeOf(client.DB{}.B),
+	}
+	blacklist := map[string]struct{}{
```
Comment (tamird) [inline client/db_test.go]: let's add a sanity check that these methods all exist so this doesn't rot when methods are removed

Comment (petermattis) [inline client/db_test.go]: Done. PTAL.

**Comment by tamird** [comment]
LGTM

**Pull request merged**